### PR TITLE
Smtp server detection 1125 v17

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/1125 preliminary work

Describe changes:
- protocoldetect: use stricter max depth
- pop3: add detection

Modifies #8804 by needed rebase

Summing up :
- Protocol detection has many bugs (see first commits)
- Especially, eve.json `stats` field about flows are not matching the count of `flow` with app_proto because of so many corner cases
- QA baseline is wrong because of counting pop3 flows as FTP ones (because of `USER` pattern matched)
-  QA baseline is wrong because of counting IRC flows on port 5432 as pgsql because pgsql probing parser to client accepts anything

@victorjulien how should this be handled ? Create many tickets and find the right priority order to deal with them ?
Should it begin with just commit f27a6f1480aab938b5511a6f96f8d0a1873ff710 but using port 110 instead of POP3 ? That would solve QA wrongly tagging POP3 flows as FTP

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 

ERROR: QA failed on IPS_AFP_drop_chk.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.tx.ftp | 646 | 187 | 28.95% |
| .app_layer.error.ftp.gap | 2 | 0 | - |
| .app_layer.error.ftp.parser | 2 | 0 | - |
| .ftp.memuse | 348 | 3 | 0.86% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.pgsql | 0 | 55 | - |
| | IPS_AFP_stats_chk | | |
| .flow.end.state.established | 583199 | 550799 | 94.44% |
| .flow.end.tcp_state.established | 201960 | 169560 | 83.96% |
| .app_layer.flow.ftp | 33480 | 1080 | 3.23% |
| .app_layer.tx.ftp | 131760 | 2160 | 1.64% |
| .app_layer.error.ftp.parser | 32400 | 0 | - |
| | TREX_GENERIC_stats_chk | | |
| .app_layer.flow.ftp | 18444 | 0 | - |
| .app_layer.tx.ftp | 73776 | 0 | - |
| .app_layer.error.ftp.parser | 18444 | 0 | - | 

Pipeline 13821

File: src/app-layer-detect-proto.c [hunk 1]
```diff
@@ -287,6 +287,10 @@ static inline int PMGetProtoInspect(AppLayerProtoDetectThreadCtx *tctx,
     uint8_t pm_results_bf[(ALPROTO_MAX / 8) + 1];
     memset(pm_results_bf, 0, sizeof(pm_results_bf));
 
+    // Do not take pm_ctx->pp_max_len of all probing parsers,
+    // but only the the probing parsers which matched a pattern.
```
Comment (jufajardini) [inline src/app-layer-detect-proto.c:291]: nit: remove extra `the`

Comment (catenacyber) [reply src/app-layer-detect-proto.c]: Thanks, done (still waiting for my big questions to be answered)

**Comment by suricata-qa** [comment]
WARNING:
 

ERROR: QA failed on IPS_AFP_drop_chk.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.tx.ftp | 646 | 187 | 28.95% |
| .app_layer.error.ftp.gap | 2 | 0 | - |
| .app_layer.error.ftp.parser | 2 | 0 | - |
| .ftp.memuse | 348 | 3 | 0.86% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.pgsql | 0 | 55 | - |
| | IPS_AFP_stats_chk | | |
| .flow.end.state.established | 583199 | 550799 | 94.44% |
| .flow.end.tcp_state.established | 201960 | 169560 | 83.96% |
| .app_layer.flow.ftp | 33480 | 1080 | 3.23% |
| .app_layer.tx.ftp | 131760 | 2160 | 1.64% |
| .app_layer.error.ftp.parser | 32400 | 0 | - |
| | TREX_GENERIC_stats_chk | | |
| .app_layer.flow.ftp | 18444 | 0 | - |
| .app_layer.tx.ftp | 73776 | 0 | - |
| .app_layer.error.ftp.parser | 18444 | 0 | - | 

Pipeline 14077

**Comment by suricata-qa** [comment]
WARNING:
 

ERROR: QA failed on IPS_AFP_drop_chk.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.tx.ftp | 646 | 187 | 28.95% |
| .app_layer.error.ftp.gap | 2 | 0 | - |
| .app_layer.error.ftp.parser | 2 | 0 | - |
| .ftp.memuse | 348 | 3 | 0.86% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.pgsql | 0 | 55 | - |
| | IPS_AFP_stats_chk | | |
| .flow.end.state.established | 583199 | 550799 | 94.44% |
| .flow.end.tcp_state.established | 201960 | 169560 | 83.96% |
| .app_layer.flow.ftp | 33480 | 1080 | 3.23% |
| .app_layer.tx.ftp | 131760 | 2160 | 1.64% |
| .app_layer.error.ftp.parser | 32400 | 0 | - |
| | TREX_GENERIC_stats_chk | | |
| .app_layer.flow.ftp | 18444 | 0 | - |
| .app_layer.tx.ftp | 73776 | 0 | - |
| .app_layer.error.ftp.parser | 18444 | 0 | - | 

Pipeline 14077

**Comment by catenacyber** [comment]
Continues in https://github.com/OISF/suricata/pull/9500

**Pull request closed without merging**